velox/exec/tests/TableWriteTest.cpp

```diff
@@ -610,3 +644,111 @@ TEST_F(TableWriteTest, writeNoFile) {
   execute(plan, std::make_shared<core::QueryCtx>(executor_.get()));
   ASSERT_TRUE(fs::is_empty(outputDirectory->path));
 }
+
+TEST_F(TableWriteTest, immutablePartitionsExistingTableErrorWhenEnabled) {
+  auto input = makeVectors(rowType_, 10, 10);
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = createInsertPlan(
+      rowType_,
+      input,
+      connector::hive::LocationHandle::TableType::kExisting,
+      outputDirectory->path);
+
+  VELOX_ASSERT_THROW(
+      AssertQueryBuilder(plan)
+          .connectorConfig(
+              kHiveConnectorId,
+              HiveConfig::kImmutablePartitions,
+              folly::to<std::string>("true"))
+          .copyResults(pool()),
+      "Unpartitioned Hive tables are immutable.");
+}
+
+TEST_F(TableWriteTest, immutablePartitionsNewTableEnabled) {
+  // When table type is NEW, we always return UpdateMode::kNew, no exception
+  // should be thrown.
+  auto input = makeVectors(rowType_, 10, 10);
+  auto outputDirectory = TempDirectoryPath::create();
+  auto plan = createInsertPlan(
+      rowType_,
+      input,
+      connector::hive::LocationHandle::TableType::kNew,
+      outputDirectory->path);
+
+  auto result = AssertQueryBuilder(plan)
+                    .connectorConfig(
+                        kHiveConnectorId,
+                        HiveConfig::kImmutablePartitions,
+                        folly::to<std::string>("true"))
+                    .copyResults(pool());
+
+  assertEqualResults(
+      {makeRowVector({makeConstant<int64_t>(100, 1)})}, {result});
+}
+
+TEST_F(TableWriteTest, insertZeroRows) {
+  auto input = makeVectors(rowType_, 10, 0);
```

Comment:
This and the below test are exactly the same except in the number of rows. Please can you abstract a common function that parameterizes based on the number of rows and use it in your tests.

Reply:
Updated.